test/es-module/test-esm-throw-undefined.mjs

```diff
@@ -0,0 +1,15 @@
+// Flags: --experimental-modules
+/* eslint-disable node-core/required-modules */
+import common from '../common/index.js';
+import assert from 'assert';
+
+async function doTest() {
+  await assert.rejects(
+    async () => {
+      await import('../fixtures/es-module-loaders/throw-undefined');
+    },
+    (e) => e === undefined
+  );
```

Comment:
Nit:

```js
await assert.rejects(
  import('../fixtures/es-module-loaders/throw-undefined'),
  (e) => e === undefined);
```

The extra arrow function doesn't seem needed.

Reply:
I just followed the example here:
https://github.com/nodejs/node/blob/master/doc/api/assert.md

You mean, the first parameter can be a promise (not a function)? I don't know, but I think mine is more formal :)

Reply:
Oh oops, never mind that.